src/dpl/src/CheckPlacement.cpp

```diff
@@ -175,115 +169,95 @@ void Opendp::processViolationsPtree(boost::property_tree::ptree& entry,
 
-      xMin = (overlap_rect.xMin() + core.xMin()) / dbUnits;
-      yMin = (overlap_rect.yMin() + core.yMin()) / dbUnits;
-      xMax = (overlap_rect.xMax() + core.xMin()) / dbUnits;
-      yMax = (overlap_rect.yMax() + core.yMin()) / dbUnits;
+      xMin = overlap_rect.xMin() + core.xMin();
+      yMin = overlap_rect.yMin() + core.yMin();
+      xMax = overlap_rect.xMax() + core.xMin();
+      yMax = overlap_rect.yMax() + core.yMin();
 
-      ptree overlap_source;
-      overlap_source.put("type", "inst");
-      overlap_source.put("name", o_cell->name());
-      sources.push_back(std::make_pair("", overlap_source));
+      marker->addSource(o_cell->db_inst_);
     }
-    shape.put("x", xMin);
-    shape.put("y", yMin);
-    shapes.push_back(std::make_pair("", shape));
-    shape.clear();
-    shape.put("x", xMax);
-    shape.put("y", yMax);
-    shapes.push_back(std::make_pair("", shape));
-
-    source.put("type", "inst");
-    source.put("name", failure->name());
-    sources.push_back(std::make_pair("", source));
-
-    violation.put("type", "box");
-    violation.add_child("shape", shapes);
-    violation.add_child("sources", sources);
-
-    violations.push_back(std::make_pair("", violation));
+    marker->addShape(Rect{xMin, yMin, xMax, yMax});
+    marker->addSource(failure->db_inst_);
   }
-  entry.add_child("violations", violations);
 }
 
-void Opendp::writeJsonReport(const string& filename,
-                             const vector<Cell*>& placed_failures,
-                             const vector<Cell*>& in_rows_failures,
-                             const vector<Cell*>& overlap_failures,
-                             const vector<Cell*>& one_site_gap_failures,
-                             const vector<Cell*>& site_align_failures,
-                             const vector<Cell*>& region_placement_failures,
-                             const vector<Cell*>& placement_failures_)
+void Opendp::saveFailures(const vector<Cell*>& placed_failures,
+                          const vector<Cell*>& in_rows_failures,
+                          const vector<Cell*>& overlap_failures,
+                          const vector<Cell*>& one_site_gap_failures,
+                          const vector<Cell*>& site_align_failures,
+                          const vector<Cell*>& region_placement_failures,
+                          const vector<Cell*>& placement_failures)
 {
-  std::ofstream json_file(filename);
-  if (!json_file.is_open()) {
-    logger_->error(DPL, 40, "Failed to open file {} for writing.", filename);
+  if (placed_failures.empty() && in_rows_failures.empty()
+      && overlap_failures.empty() && one_site_gap_failures.empty()
+      && site_align_failures.empty() && region_placement_failures.empty()
+      && placement_failures.empty()) {
+    return;
   }
-  try {
-    using boost::property_tree::ptree;
-    ptree root, drcArray;
 
+  auto* tool_category = odb::dbMarkerCategory::createOrGet(block_, "DPL");
```

Comment:
You might want to use `createOrReplace` if I run the check placement command twice and have placement failures at one time, but later I only have in row failures it would appear that both will be reported.

Reply:
I used
```
  odb::dbMarkerCategory* tool_category
      = odb::dbMarkerCategory::createOrGet(getBlock(), "PSM");
```
as a template.  Is this incorrect too?

Reply:
also in connect.cpp and TritonRoute.cpp

Reply:
in PSM the net markers are in different categories so that get handled there. It's possible the behavior is not correct in the other places. (in PDN I believe I destroy the old markers first)